src/Microsoft.Extensions.ProjectModel.MsBuild.Sources/MsBuildProjectContextBuilder.cs

```diff
@@ -37,7 +42,7 @@ public IProjectContext Build()
                     "msbuild",
                     _projectPath,
                     $"/t:EvaluateProjectInfoForCodeGeneration", 
-                    $"/p:TargetFramework={_targetFramework.GetShortFolderName()};OutputFile={tmpFile}"
+                    $"/p:OutputFile={tmpFile};CodeGenerationTargetLocation={_targetLocation}"
```

Comment:
Ahhh, face palm worthy. I reversed the order of arguments when following the code path... doh!